### ISSUE TITLE
Reduce Search "Churn" on Network Page

### ISSUE THREAD

**Issue by DaleMcGrew** [wevote/WebApp#1710]
On the network page, a voter can search for organizations to follow. When you type in the search box, it starts searching as you are typing. What can happen is that the API call can cause "churn" as the responses come in and try to get loaded on the page. I think there are a few things we can do to reduce this problem:

- [ ] Do not start instant search until there are at least 3 characters
- [ ] Limit the requested search results to 50
- [ ] Increase the delay after a key stroke before requesting from the API
- [ ] Note also that deleting the contents of the search box should make a fresh call to the voterGuidesToFollowRetrieve API in order to reset the search page to the original listing.

This is the API Call used:
https://api.wevoteusa.org/apis/v1/voterGuidesToFollowRetrieve/?google_civic_election_id=0&start_retrieve_at_this_number=0&maximum_number_to_retrieve=25&search_string=pla&add_voter_guides_not_from_election=false

![screen shot 2018-10-06 at 4 15 23 am](https://user-images.githubusercontent.com/7756031/46570817-24995d80-c91f-11e8-903e-22ba327b2ddc.png)

**Comment by gj262** [comment]
I'll take a look at this if no one has started on it. I have some time today.
Cheers,
-Gavin

**Comment by DaleMcGrew** [comment]
Hi @gj262 I've been working on this -- may I suggest another Issue? I will tag you in another possible issue. Would you be willing to email me so we can connect by email? I am: Dale.McGrew@WeVote.US

**Comment by DaleMcGrew** [comment]
On further thought, @gj262 would you want to tackle optimizing this search within the WebApp? It will require a deeper dive than the prior issues you've worked on, and I'm happy to help if you have any questions on a Google hangout.
